javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/instrumentation/indy/AdviceTransformer.java

```diff
@@ -554,105 +583,115 @@ public void visitCode() {
   }
 
   private static void instrument(
-      TransformationContext context, MethodNode methodNode, ClassVisitor classVisitor) {
+      TransformationContext context,
+      MethodNode methodNode,
+      ClassVisitor classVisitor,
+      boolean justDelegateAdvice) {
+
     String originalDescriptor = methodNode.desc;
     String[] exceptionsArray = methodNode.exceptions.toArray(new String[0]);
 
-    List<OutputArgument> writableArguments = getWritableArguments(methodNode);
-    OutputArgument writableReturn = getWritableReturnValue(methodNode);
-    OutputArgument enterArgument = getEnterArgument(methodNode);
-    List<AdviceLocal> adviceLocals = getLocals(methodNode);
-    boolean isEnterAdvice = isEnterAdvice(methodNode);
-    boolean isExitAdvice = isExitAdvice(methodNode);
-    Type returnType = Type.getReturnType(methodNode.desc);
-
-    // currently we don't support rewriting enter advice returning a primitive type
-    if (isEnterAdvice
-        && !(returnType.getSort() == Type.VOID
-            || returnType.getSort() == Type.OBJECT
-            || returnType.getSort() == Type.ARRAY)) {
-      context.disableReturnTypeChange();
-    }
-    // context is shared by enter and exit advice, if entry advice was rejected don't attempt to
-    // rewrite usages of @Advice.Enter in the exit advice
-    if (!context.canChangeReturnType()) {
-      enterArgument = null;
-    }
+    if (!justDelegateAdvice) {
```

Comment:
instead of having an if block that surrounds everything could extract the transform that changes the inline advice to delegating advice into helper method and call it here or perhaps even call it instead of `instrument` when only changing the advice type is desired

Reply:
Thanks, I've made the changes.